sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1902,249 @@ def transform(self, X):
+    array([[ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.]])
+
+    See also
+    --------
+    https://blogs.technet.microsoft.com/machinelearning/2015/02/17/big-learning-made-easy-with-counts/      # noqa
+    https://msdn.microsoft.com/en-us/library/azure/dn913056.aspx
+    """
+
+    def __init__(self, inclusion='all'):
+        self.inclusion = inclusion
+
+    @staticmethod
+    def _check_params(inclusion=None):
+        if inclusion is None:
+            raise ValueError("Inclusion cannot be none")
+        if (inclusion is not None and (type(inclusion) == str and
+            inclusion != "all" and inclusion != "each") and
+                not CountFeaturizer._valid_data_type(inclusion)):
+            raise ValueError("Illegal data type in inclusion")
+
+    @staticmethod
+    def _valid_data_type(type_check):
+        return type(type_check) == np.ndarray or type(type_check) == list
+
+    def _get_inclusion_used(self):
+        if type(self.inclusion) == str and self.inclusion == "all":
+            return np.array([range(self.col_num_X_)])
+        elif type(self.inclusion) == str and self.inclusion == "each":
+            return np.array([[i] for i in range(self.col_num_X_)])
+        elif CountFeaturizer._valid_data_type(self.inclusion):
+            inclusion_used = np.array(self.inclusion)
+            if len(inclusion_used.shape) == 1:
+                return np.array([inclusion_used])
+            else:
+                return inclusion_used
+
+    def fit(self, X, y=None):
+        """Fits the CountFeaturizer to X, y
+
+        Stores the counts for each example X, conditional on y
+        Both X and y must be appropriately reshaped to a 2D list
+
+        Parameters
+        ----------
+        X : array
+            The data set to learn the counts from, conditional to 'y'
+            X must not be 1 dimensional
+
+        y : array-like, optional
+            If provided, a separate column is output for each value of 'y',
+            counting the occurences of 'X' conditioned on that 'y' value
+
+        Returns
+        -------
+        self
+        """
+
+        CountFeaturizer._check_params(inclusion=self.inclusion)
+
+        if y is not None:
+            X, y = check_X_y(X, y, multi_output=True)
+            if len(y.shape) == 1:
+                self.col_num_Y_ = 1
+            else:
+                self.col_num_Y_ = len(y[0])
+        else:
+            X = check_array(X)
+            self.col_num_Y_ = 1
+
+        len_data = len(X)
+        self.col_num_X_ = len(X[0])
+        self.count_cache_ = _get_count_dict_3()
+        self.classes_ = [set() for i in range(self.col_num_Y_)]
+
+        inclusion_used = self._get_inclusion_used()
+
+        for inclusion_i in range(len(inclusion_used)):
+            if y is not None:
```

Comment:
Changed to `np.zeros(len(X))`
Was unsure about it before because it would create `len(X)` additional elements in memory that wouldn't really have that much use, but it certainly did make the code a lot cleaner.